scripts/generate_data_in_parallel.sh

```diff
@@ -1,26 +1,31 @@
 #!/bin/bash
 
-# gracefully exit
-trap "cleanup" SIGINT SIGTERM EXIT
-
-# Usage: generate_data_in_parallel.sh <relative config_path> <num_files_per_process> [<config_overrides> <num_processes> <num_workers_per_process>]
-CONFIG_PATH=$1
-N_FILES_PER_PROCESS=$2
-CONFIG_OVERRIDES=${3:-""}
-NUM_PROCESSES=${4:-32}
-N_WORKERS_PER_PROCESS=${5:-0}
-
-# Error if CONFIG_PATH, N_FILES_PER_PROCESS not specified
-if [ -z "$CONFIG_PATH" ] || [ -z "$N_FILES_PER_PROCESS" ]; then
-    echo "Usage: generate_data_in_parallel.sh <relative config_path> <num_files_per_process> <num_processes (default: 32)> <num_workers_per_process (default: 0)>"
+
+# Gracefully exit when interrupted
+cleanup() {
+    echo "Ending background processes"
+    for pid in "${pids[@]}";
+    do
+        kill -15 $pid
+    done
+}
+
+# gracefully exit on these conditions
+trap "cleanup" SIGINT SIGTERM
+
+NUM_PROCESSES=${1:-""}
+CONFIG_OVERRIDES=${2:-""}
+
+# Error if NUM_PROCESS not specified
+if [ -z "$NUM_PROCESSES" ]; then
+    echo "Usage: generate_data_in_parallel.sh <num_processes> [<config_overrides>]"
```

Comment:
I think this interface is easier to understand/use: the only mandatory argument, num_processes, is used by this script and not passed on the bliss/hydra. The rest of the arguments pass through to bliss/hydra.

Reply:
Agreed